packages/core/tests/test_integration.py

```diff
@@ -0,0 +1,194 @@
+import pickle
+
+import pytest
+from bag3d.common.resources import gdal
+from bag3d.common.resources.executables import geoflow, roofer, tyler
+from bag3d.common.resources.files import file_store
+from bag3d.common.resources.temp_until_configurableresource import (
+    EXE_PATH_GEOF, EXE_PATH_ROOFER_CROP, EXE_PATH_TYLER, EXE_PATH_TYLER_DB,
+    FLOWCHART_PATH_RECONSTRUCT)
+from bag3d.common.types import PostgresTableIdentifier
+from bag3d.core.assets import export, reconstruction
+from bag3d.core.assets.export.tile import reconstruction_output_tiles_func
+from bag3d.core.assets.input import RECONSTRUCTION_INPUT_SCHEMA
+from bag3d.core.jobs import job_nl_export, job_nl_reconstruct
+from dagster import (AssetKey, Definitions, ExecuteInProcessResult, IOManager,
+                     Output, SourceAsset, load_assets_from_package_module)
+
+
+def mock_reconstruction_input(reconstruction_input):
+    class MockIOManager(IOManager):
+        def load_input(self, context):
+            new_table = PostgresTableIdentifier(
+                RECONSTRUCTION_INPUT_SCHEMA, "reconstruction_input"
+            )
+            return new_table
+
+        def handle_output(self, context, obj):  # pragma: no cover
+            raise NotImplementedError()
+
+    return SourceAsset(
+        key=AssetKey(["input", reconstruction_input]),
+        io_manager_def=MockIOManager(),
+    )
+
+
+def mock_tiles(tiles):
+    class MockIOManager(IOManager):
+        def load_input(self, context):
+            new_table = PostgresTableIdentifier(RECONSTRUCTION_INPUT_SCHEMA, "tiles")
+            return new_table
+
+        def handle_output(self, context, obj):  # pragma: no cover
+            raise NotImplementedError()
+
+    return SourceAsset(
+        key=AssetKey(["input", tiles]),
+        io_manager_def=MockIOManager(),
+    )
+
+
+def mock_index(index):
+    class MockIOManager(IOManager):
+        def load_input(self, context):
+            new_table = PostgresTableIdentifier(RECONSTRUCTION_INPUT_SCHEMA, "index")
+            return new_table
+
+        def handle_output(self, context, obj):  # pragma: no cover
+            raise NotImplementedError()
+
+    return SourceAsset(
+        key=AssetKey(["input", index]),
+        io_manager_def=MockIOManager(),
+    )
+
+
+def mock_regular_grid_200m(regular_grid_200m):
+    class MockIOManager(IOManager):
+        def load_input(self, context):
+            new_table = PostgresTableIdentifier("ahn", "regular_grid_200m")
+            return new_table
+
+        def handle_output(self, context, obj):  # pragma: no cover
+            raise NotImplementedError()
+
+    return SourceAsset(
+        key=AssetKey(["ahn", regular_grid_200m]),
+        io_manager_def=MockIOManager(),
+    )
+
+def mock_export_index(export_index, test_data_dir):
```

Comment:
I found this way of mocking upstream assets in the docs. Not sure if there is a better way.